Core/source/core/pgp-msg.ts

```diff
@@ -173,19 +174,19 @@ export class PgpMsg {
         // .. which is not really an issue - an attacker that can append signatures
         // could have also just slightly changed the message, causing the same experience
         // .. so for now #wontfix unless a reasonable usecase surfaces
-        sig.match = (sig.match === true || sig.match === null) && await verifyRes.verified;
         if (!sig.signer) {
           // todo - currently only the first signer will be reported.
           // Should we be showing all signers? How common is that?
           sig.signer = await PgpKey.longid(verifyRes.keyID.bytes);
         }
+        sig.match = (sig.match === true || sig.match === null) && await verifyRes.verified;
```

Comment:
`verifyRes.verified` throws on invalid signature in `openpgpjs 5`. This lead to `sig.signer` to be `null`, so I moved `sig.match` below.

Reply:
And I suppose we had some test that expected `sig.signer` to be set even when it's a bad signature? Could you point me to it?

Reply:
Yes, we use `sig.signer` when detecting type of signature error - https://github.com/FlowCrypt/flowcrypt-ios/blob/master/FlowCrypt/Functionality/Mail%20Provider/Message%20Provider/MessageService.swift#L256

Previously openpgp didn't throw error when `sig.match = nil`, but in v5 this logic changed and now we need manually check error type for `.missingPubKey` error. (old code - https://github.com/FlowCrypt/flowcrypt-ios/blob/e89baece51bcca00b8ad4c58193541a8ab863d4f/FlowCrypt/Functionality/Mail%20Provider/Message%20Provider/MessageService.swift#L259)

This change lead to fail of `CheckAllEmailSignatureCases` test - https://github.com/FlowCrypt/flowcrypt-ios/blob/master/appium/tests/specs/mock/inbox/CheckAllEmailSignatureCases.spec.ts#L64